internal/pkg/scorecard/scorecard.go

```diff
@@ -171,37 +171,44 @@ func runTests() ([]scapiv1alpha1.ScorecardOutput, error) {
 			return nil, err
 		}
 
-		// Create a temporary CR manifest from metadata if one is not provided.
-		crJSONStr, ok := csv.ObjectMeta.Annotations["alm-examples"]
-		if ok && viper.GetString(CRManifestOpt) == "" {
-			var crs []interface{}
-			if err = json.Unmarshal([]byte(crJSONStr), &crs); err != nil {
-				return nil, err
-			}
-			// TODO: run scorecard against all CR's in CSV.
-			cr := crs[0]
-			crJSONBytes, err := json.Marshal(cr)
-			if err != nil {
-				return nil, err
-			}
-			crYAMLBytes, err := yaml.JSONToYAML(crJSONBytes)
-			if err != nil {
-				return nil, err
-			}
-			crFile, err := ioutil.TempFile("", "cr.yaml")
-			if err != nil {
-				return nil, err
-			}
-			if _, err := crFile.Write(crYAMLBytes); err != nil {
-				return nil, err
-			}
-			viper.Set(CRManifestOpt, crFile.Name())
-			defer func() {
-				err := os.Remove(viper.GetString(CRManifestOpt))
+		if len(viper.GetStringSlice(CRManifestOpt)) == 0 {
+			// Create a temporary CR manifest from metadata if one is not provided.
+			if crJSONStr, ok := csv.ObjectMeta.Annotations["alm-examples"]; ok {
+				var crs []interface{}
+				if err = json.Unmarshal([]byte(crJSONStr), &crs); err != nil {
+					return nil, err
+				}
+				// TODO: run scorecard against all CR's in CSV.
+				cr := crs[0]
+				crJSONBytes, err := json.Marshal(cr)
 				if err != nil {
-					log.Errorf("Could not delete temporary CR manifest file: (%v)", err)
+					return nil, err
 				}
-			}()
+				crYAMLBytes, err := yaml.JSONToYAML(crJSONBytes)
+				if err != nil {
+					return nil, err
+				}
+				crFile, err := ioutil.TempFile("", "*.cr.yaml")
+				if err != nil {
+					return nil, err
+				}
+				if _, err := crFile.Write(crYAMLBytes); err != nil {
+					return nil, err
+				}
+				viper.Set(CRManifestOpt, []string{crFile.Name()})
+				defer func() {
+					for _, f := range viper.GetStringSlice(CRManifestOpt) {
+						if err := os.Remove(f); err != nil {
+							log.Errorf("Could not delete temporary CR manifest file: (%v)", err)
+						}
+					}
+				}()
+			} else {
+				return nil, errors.New("cr-manifest config option must be set if CSV has no metadata.annotations['alm-examples']")
+			}
+		} else {
+			// TODO: run scorecard against all CR's in CSV.
+			viper.Set(CRManifestOpt, []string{viper.GetStringSlice(CRManifestOpt)[0]})
```

Comment:
In the previous version of the code, it looks like we support multiple CRs with `--olm-deployed` if passed via `--cr-manifest`. This version now takes only the first CR from the list. Am I understanding that correctly?

Other than properly parsing and handling the flags or annotation, is there a reason mutliple CRs are not supported with `--olm-deployed`?

Also, in the cases that we only select the first CR from the list, should we log a message to tell the user that multiple CRs are not supported in that case?

Reply:
To test multiple CR's, the scorecard cleans up the resources it created in between each CR test. This currently doesn't work if OLM deployed the operator because there required cleanup (and creation) logic isn't present. The user must manage creation/deletion when using `--olm-deployed`. I'm working on that in another branch. In the mean time, the scorecard should at least work for one CR using `--olm-deployed`.

I'll add a log statement for the case that multiple CR's are passed/in `alm-examples`.